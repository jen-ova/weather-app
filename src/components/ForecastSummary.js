import React from "react";
import PropTypes from "prop-types";
// import WeatherIcon from "react-icons-weather";

const ForecastSummary = (props) => {
  const { date, description, icon, temperature } = props;

  return (
    <div className="forecast-summary">
      <div className="forecast-summary__date">{date}</div>
      <div className="forecast-summary__description">{description}</div>
      <div className="forecast-summary__icon">
        {icon}
        {/* <WeatherIcon name="owm" iconId={icon} /> */}
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}
        &deg;C
      </div>
    </div>
  );
};

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};

export default ForecastSummary;
