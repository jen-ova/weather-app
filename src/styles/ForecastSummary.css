.forecast-summary__date {
    font-size: 1rem;
}

.forecast-summary__icon {
    margin: 0.75rem 0 0.75rem 0;
    font-size: 3.5rem;
}