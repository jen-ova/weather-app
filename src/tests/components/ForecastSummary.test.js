import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1525046400000,
    description: "stubDescription",
    icon: "211",
    temperature: {
      min: 12,
      max: 24,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(getByText("Mon 30th Apr")).toHaveClass("forecast-summary__date");
    expect(getByText("stubDescription")).toHaveClass(
      "forecast-summary__description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("24Â°C")).toHaveClass("forecast-summary__temperature");
  });
});
