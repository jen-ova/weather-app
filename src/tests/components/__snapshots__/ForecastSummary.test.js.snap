// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      1111111
    </div>
    <div
      class="forecast-summary__description"
    >
      stubDescription
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-thunderstorm"
        name="owm"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      24Â°C
    </div>
  </div>
</DocumentFragment>
`;
